base_comment_template/models/ir_model.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 NextERP Romania SRL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class IrModel(models.Model):
+    _inherit = "ir.model"
+
+    is_comment_template = fields.Boolean(
+        string="Comment Template",
+        default=False,
+        help="Whether this model supports in reports to add comment templates.",
+    )
+
+    def _reflect_model_params(self, model):
+        vals = super(IrModel, self)._reflect_model_params(model)
+        vals["is_comment_template"] = issubclass(
+            type(model), self.pool["comment.template"]
+        )
+        return vals
+
+    @api.model
+    def _instanciate(self, model_data):
```

Comment:
I've seen in unit test that a res.users fake model is created, in suppose if we add a test that instantiate that model this code would be called, isn't it ?

also I'm not sure what that code is supposed to do, it's look like a bit too magic I wondering if a mixin would do the trick ?

Reply:
Yest, exactly.
These addon need to test if some model inherit comment template and everything work fine. 
We need add some code to ir_model because only with inherit comment.template we can select model in the templates.
Maybe some documentation about fake_models https://pypi.org/project/odoo-test-helper/ help you.